src/main/scala/ai/privado/passes/DBTParserPass.scala

```diff
@@ -0,0 +1,462 @@
+import com.typesafe.config.*
+
+import scala.xml.XML
+import com.github.wnameless.json.flattener.JsonFlattener
+import io.circe.yaml.parser
+import org.yaml.snakeyaml.{LoaderOptions, Yaml}
+import org.yaml.snakeyaml.nodes.{MappingNode, Node, NodeTuple, ScalarNode, SequenceNode}
+
+import scala.collection.immutable.HashMap
+import scala.collection.mutable.ArrayBuffer
+import scala.jdk.CollectionConverters.*
+import ai.privado.model._
+import ai.privado.tagger.PrivadoParallelCpgPass
+import com.github.javaparser.utils.ProjectRoot
+import org.yaml.snakeyaml.constructor.SafeConstructor
+
+import scala.collection.mutable.ListBuffer
+import scala.util.{Failure, Success, Try}
+import java.nio.file.{Files, Path, Paths}
+import scala.util.control.Breaks.*
+import scala.util.matching.Regex
+
+object FileExtensions {
+  val YAML = ".yaml"
+  val YML  = ".yml"
+}
+
+class DBTParserPass(cpg: Cpg, projectRoot: String, ruleCache: RuleCache) extends PrivadoParallelCpgPass[String](cpg) {
+
+  val logger = LoggerFactory.getLogger(getClass)
+
+  override def generateParts(): Array[String] = {
+    getConfigFiles(
+      projectRoot,
+      Set(FileExtensions.YAML, FileExtensions.YML),
+      Set("dbt_project[.]yaml", "dbt_project[.]yml")
+    ).toArray
+  }
+
+  override def runOnPart(builder: DiffGraphBuilder, projectFile: String): Unit = {
+    logger.debug(f"Processing DBT Project: $projectFile")
+    getYAML(projectFile) match {
+      case Success(projectData) => {
+        val projectName         = projectData.get("name").asInstanceOf[String]
+        val profileName         = projectData.get("profile").asInstanceOf[String]
+        val modelsDirectoryName = getModelsDirectoryName(projectData)
+
+        // extract db information from profile
+        val (profileFile, dbKey, dbName, dbHost, dbPlatform) = getDatabaseInfo(projectFile, profileName) match {
+          case Success(value) => value
+          case Failure(err) =>
+            logger.error(f"error while getting database info: ${projectFile}: ${err}")
+            ("", "", "DBT", "", "")
+        }
+
+        // get all models from the model directory
+        val models = getModels(projectFile, modelsDirectoryName) match {
+          case Success(value) => value
+          case Failure(err) =>
+            logger.error(f"error while getting models for profile: ${projectFile}: ${err}")
+            Array[java.util.Map[String, Any]]()
+        }
+
+        // transform models into expected schema format
+        val schema = transformModelsToSchema(projectName, dbPlatform, models)
+        // create rule and add metadata for dbt storage sink
+        val ruleInfo = createDatabaseSinkRule(projectName, dbName, dbPlatform, dbHost, schema)
+
+        // create and tag database node
+        val dbNode = createAndTagDatabaseNode(builder, ruleInfo, projectFile, projectName, profileFile, dbName, dbKey)
+        // create sql node
+        val sqlNode = createSQLNode(builder, dbNode)
```

Comment:
Rename the function to something like `createSQLQueryNode` and also update the above comment